lib/mongoid/document.rb

```diff
@@ -228,6 +215,34 @@ def becomes(klass)
 
     private
 
+    # Does the construction of a document.
+    #
+    # @param [ Hash ] attrs The attributes to set up the document with.
+    # @param [ true | false ] defer_callbacks Flag specifies whether callbacks
+    #   should be run.
+    #
+    # @return [ Document ] A new document.
+    def construct_document(attrs = nil, defer_callbacks = false)
```

Comment:
I would prefer `run_callbacks` (positive and matches the method that eventually will be called), and I would like it to be a keyword argument. If this method is always called with the first argument (`attrs`), the first argument should be required. Finally, even though this method is private, it should still be marked `@api private` because application models include `Mongoid::Document` and thus are able to invoke private methods.

Reply:
Disagree. As per my previous comment, `run_callbacks` (previously `execute_callbacks`) is not a good name because `run_callbacks = false` implies that callbacks will never be run.

Personally I also prefer default variables to be `nil` or `false`.

Reply:
I have flipped this flag and renamed it (back) to `execute_callbacks`